Index: jdiff.cpp
===================================================================
--- jdiff.cpp	(revision 70)
+++ jdiff.cpp	(working copy)
@@ -12,14 +12,33 @@
 
 #define J_LOG_SOURCE "jdiff.exe"
 
+// jedit.exe can be build as either a "windows app"
+// or a console app.  Comment this out to compile
+// as a console app.
+#define BUILD_WIN32_APP
+
 const char* USAGE = 
 "Usage: jdiff.exe [-h] file1 file2\n";
 
 
+#ifdef BUILD_WIN32_APP
+int 
+WINAPI WinMain(HINSTANCE hInstance,
+			   HINSTANCE hPrevInstance,
+			   LPSTR lpCmdLine,
+			   int nCmdShow)
+#else
 int
-main(int argc, 
+main(int argc,
 	 char** argv)
+#endif
 {
+	#ifdef BUILD_WIN32_APP
+	// avoid having to parse lpCmdLine if
+	// compiling as windows app
+	int argc = __argc;
+	char** argv = __argv;
+	#endif
 	LaunchConfig config;
 	char file1[MAX_PATH] = {'\0'};
 	char file2[MAX_PATH] = {'\0'};
@@ -33,12 +52,20 @@
 					 strcmp(argv[1], "/?") == 0 ||
 					 strcmp(argv[1], "--help") == 0))
 	{
+#ifdef BUILD_WIN32_APP
+		MessageBox(NULL, USAGE, "jDiff Usage", MB_OK);
+#else
 		printf(USAGE);
+#endif
 		exit(0);
 	}
 	else if(argc != 3)
 	{
+#ifdef BUILD_WIN32_APP
+		MessageBox(NULL, USAGE, "jDiff Usage", MB_OK);
+#else
 		printf(USAGE);
+#endif
 		exit(2);
 	}
 
