Index: org/gjt/sp/jedit/gui/DockableWindowFactory.java
===================================================================
--- org/gjt/sp/jedit/gui/DockableWindowFactory.java	(revision 11145)
+++ org/gjt/sp/jedit/gui/DockableWindowFactory.java	(working copy)
@@ -167,11 +167,16 @@
 	} //}}}
 
 	//{{{ getDockableWindowIterator() method
-	Iterator<Window> getDockableWindowIterator()
+	public Iterator<Window> getDockableWindowIterator()
 	{
 		return dockableWindowFactories.values().iterator();
 	} //}}}
 
+	public Window getDockableWindowFactory(String name)
+	{
+		return dockableWindowFactories.get(name);
+	}
+	
 	//{{{ DockableListHandler class
 	class DockableListHandler extends DefaultHandler
 	{
@@ -357,7 +362,7 @@
 	} //}}}
 
 	//{{{ Window class
-	class Window
+	public class Window
 	{
 		PluginJAR plugin;
 		String name;
@@ -417,7 +422,7 @@
 		} //}}}
 
 		//{{{ createDockableWindow() method
-		JComponent createDockableWindow(View view, String position)
+		public JComponent createDockableWindow(View view, String position)
 		{
 			load();
 
Index: org/gjt/sp/jedit/gui/DockableWindowManager.java
===================================================================
--- org/gjt/sp/jedit/gui/DockableWindowManager.java	(revision 11145)
+++ org/gjt/sp/jedit/gui/DockableWindowManager.java	(working copy)
@@ -210,7 +210,7 @@
 	 * @param config A docking configuration
 	 * @since jEdit 2.6pre3
 	 */
-	public DockableWindowManager(View view, DockableWindowFactory factory,
+	public void construct(View view, DockableWindowFactory factory,
 		View.ViewConfig config)
 	{
 		setLayout(new DockableLayout());
Index: org/gjt/sp/jedit/View.java
===================================================================
--- org/gjt/sp/jedit/View.java	(revision 11145)
+++ org/gjt/sp/jedit/View.java	(working copy)
@@ -199,6 +199,16 @@
 
 	//}}}
 
+	public void setDockableWindowManager(DockableWindowManager dwm)
+	{
+		getContentPane().add(dwm,0);
+		getContentPane().remove(dockableWindowManager);
+		dockableWindowManager.close();
+		dockableWindowManager = dwm;
+		propertiesChanged();
+		getContentPane().validate();
+	}
+	
 	//{{{ getDockableWindowManager() method
 	/**
 	 * Returns the dockable window manager associated with this view.
@@ -1170,7 +1180,8 @@
 
 		setIconImage(GUIUtilities.getEditorIcon());
 
-		dockableWindowManager = new DockableWindowManager(this,
+		dockableWindowManager = new DockableWindowManager();
+		dockableWindowManager.construct(this,
 			DockableWindowFactory.getInstance(),config);
 
 		topToolBars = new JPanel(new VariableGridLayout(
